commit 7a817c45eaeb6aa93fdb5ca4d81c363b4e4218f0
Author: Frank Filz <ffilzlnx@us.ibm.com>
Date:   Sat Feb 9 07:48:48 2008 -0500

    Cleaned up some typos that were found in the various
    places in the mountd code.
    
    Signed-off-by: Steve Dickson <steved@redhat.com>
    Signed-off-by: Frank Filz <ffilzlnx@us.ibm.com>

diff -up nfs-utils-1.1.1/utils/mountd/cache.c.orig nfs-utils-1.1.1/utils/mountd/cache.c
--- nfs-utils-1.1.1/utils/mountd/cache.c.orig	2008-02-09 09:52:40.000000000 -0500
+++ nfs-utils-1.1.1/utils/mountd/cache.c	2008-02-09 09:53:45.000000000 -0500
@@ -111,7 +111,7 @@ void auth_unix_ip(FILE *f)
 	else if (client)
 		qword_print(f, *client?client:"DEFAULT");
 	qword_eol(f);
-	xlog(D_CALL, "auth_unix_ip: client %p '%s'", client, *client?client: "DEFAULT");
+	xlog(D_CALL, "auth_unix_ip: client %p '%s'", client, client?client: "DEFAULT");
 
 	if (client) free(client);
 	free(he);
@@ -516,8 +516,10 @@ void nfsd_fh(FILE *f)
 		qword_print(f, found->e_path);
 	qword_eol(f);
  out:
-	free(found_path);
-	free(he);
+	if (found_path)
+		free(found_path);
+	if (he)
+		free(he);
 	free(dom);
 	xlog(D_CALL, "nfsd_fh: found %p path %s", found, found ? found->e_path : NULL);
 	return;		
@@ -667,9 +669,9 @@ void nfsd_export(FILE *f)
 			}
 			/* If one is a CROSSMOUNT, then prefer the longest path */
 			if (((found->m_export.e_flags & NFSEXP_CROSSMOUNT) ||
-			     (found->m_export.e_flags & NFSEXP_CROSSMOUNT)) &&
+			     (exp->m_export.e_flags & NFSEXP_CROSSMOUNT)) &&
 			    strlen(found->m_export.e_path) !=
-			    strlen(found->m_export.e_path)) {
+			    strlen(exp->m_export.e_path)) {
 
 				if (strlen(exp->m_export.e_path) >
 				    strlen(found->m_export.e_path)) {
